src/me/remie/vulcan/leaguetasks/utils/LeaguePanel.java

```diff
@@ -0,0 +1,123 @@
+package me.remie.vulcan.leaguetasks.utils;
+
+import simple.hooks.simplebot.Game;
+import simple.hooks.wrappers.SimpleWidget;
+import simple.robot.api.ClientContext;
+
+public class LeaguePanel {
+    private ClientContext ctx;
+    private static final int WIDGET_ID = 657;
+
+    public LeaguePanel(final ClientContext ctx) {
+        this.ctx = ctx;
+    }
+
+    public boolean doWeNeedToClearFilters() {
+        final SimpleWidget tierDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_TIER.getWidgetId());
+        if (!isOptionSelected(tierDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget typeDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_TYPE.getWidgetId());
+        if (!isOptionSelected(typeDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget areaDropDownMenu = ctx.widgets.getWidget(WIDGET_ID,LeaguePanelFilterType.FILTER_AREA.getWidgetId());
+        if (!isOptionSelected(areaDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget completedDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_COMPLETED.getWidgetId());
+        if (!isOptionSelected(completedDropDownMenu)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean isOptionSelected(SimpleWidget widget) {
+        if (widget == null || widget.getChild(4) == null || widget.getChild(4).getText() == null) {
+            return false;
+        }
+        return widget.getChild(4).getText().equalsIgnoreCase("all");
+    }
```

Comment:
```suggestion
    private boolean isOptionSelected(SimpleWidget widget) {
        if (widget == null) {
            return false;
        }
        final SimpleWidget child = widget.getChild(4);
        if (child == null) {
            return false;
        }
        return "all".equalsIgnoreCase(child.getText());
    }
```
A cool trick we can do is we can avoid a null check on the getText of the child and instead check if all equals the text, this way if the text is `null` we'd be checking if `"all" == null`